src/lib/Zikula/Bundle/CoreBundle/Twig/Extension/PagerExtension.php

```diff
@@ -0,0 +1,399 @@
+        }
+        if ($params['display'] == 'page') {
+            $pager['pos'] = $pager['pos'] * $pager['perpage'];
+            $pager['increment'] = 1;
+        } else {
+            $pager['increment'] = $pager['perpage'];
+        }
+
+        if ($pager['pos'] < 1) {
+            $pager['pos'] = 1;
+        }
+        if ($pager['pos'] > $pager['total']) {
+            $pager['pos'] = $pager['total'];
+        }
+        unset($params['posvar']);
+
+        // number of pages
+        $pager['countPages'] = (isset($pager['total']) && $pager['total'] > 0 ? ceil($pager['total'] / $pager['perpage']) : 1);
+        if ($pager['countPages'] < 2) {
+            return '';
+        }
+
+        // current page
+        $pager['currentPage'] = ceil($pager['pos'] / $pager['perpage']);
+        if ($pager['currentPage'] > $pager['countPages']) {
+            $pager['currentPage'] = $pager['countPages'];
+        }
+
+        $templateName = (isset($params['template'])) ? $params['template'] : 'pagercss.html.twig';
+        $pager['includeStylesheet'] = isset($params['includeStylesheet']) ? $params['includeStylesheet'] : true;
+        $anchorText = (isset($params['anchorText']) ? '#' . $params['anchorText'] : '');
+        $pager['maxPages'] = (isset($params['maxpages']) ? $params['maxpages'] : 15);
+        unset($params['template']);
+        unset($params['anchorText']);
+        unset($params['maxpages']);
+
+        if (isset($params['modname'])) {
+            $pager['module'] = $params['modname'];
+        } else {
+            $module = $request->get('module', null);
+            $name   = $request->get('name', null);
+            $pager['module'] = !empty($module) ? $module : $name;
+        }
+
+        $pager['func'] = isset($params['func']) ? $params['func'] : $request->get('func', 'index');
+        $pager['type'] = isset($params['type']) ? $params['type'] : $request->get('type', 'user');
+
+        $pager['route'] = $params['route'];
+
+        $pager['args'] = array();
+        if (empty($pager['module'])) {
+            $pager['module'] = \System::getVar('startpage');
+            $starttype = \System::getVar('starttype');
+            $pager['type'] = !empty($starttype) ? $starttype : 'user';
+            $startfunc = \System::getVar('startfunc');
+            $pager['func'] = !empty($startfunc) ? $startfunc : 'index';
+
+            $startargs   = explode(',', \System::getVar('startargs'));
+            foreach ($startargs as $arg) {
+                if (!empty($arg)) {
+                    $argument = explode('=', $arg);
+                    $pager['args'][$argument[0]] = $argument[1];
+                }
+            }
+        }
+
+        //also $_POST vars have to be considered, i.e. for search results
+        $allVars = ($params['includePostVars']) ? array_merge($_POST, $_GET, $routeParams) : array_merge($_GET, $routeParams);
+        foreach ($allVars as $k => $v) {
```

Comment:
the whole method needs to be updated to eliminate BC, but didn't take to do so yet. (see `@todo` in header)